data-prepper-plugins/encryption-plugin/src/main/java/org/opensearch/dataprepper/plugins/encryption/DefaultEncryptionEngine.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.encryption;
+
+import org.opensearch.dataprepper.model.encryption.EncryptionEngine;
+import org.opensearch.dataprepper.model.encryption.EncryptionEnvelope;
+import org.opensearch.dataprepper.model.encryption.KeyProvider;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+
+public class DefaultEncryptionEngine implements EncryptionEngine {
```

Comment:
Please make all the internal class package protected to help with encapsulation.

Reply:
fixed.